lib/private/AllConfig.php

```diff
@@ -260,6 +260,10 @@ public function setUserValue($userId, $appName, $key, $value, $preCondition = nu
 		// TODO - FIXME
 		$this->fixDIInit();
 
+		if ($appName === 'core' && $key === 'email') {
```

Comment:
```suggestion
		if ($appName === 'core' && $key === 'email' && is_string($value)) {
```

Should be enough to tell psalm to be more quite :)

Reply:
Added a cast instead